hazelcast/src/main/java/com/hazelcast/quorum/QuorumCallbackImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum;
+
+import com.hazelcast.instance.MemberImpl;
+import com.hazelcast.nio.Address;
+import com.hazelcast.spi.InternalCompletableFuture;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.util.ExceptionUtil;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * QuorumCallback can be used to notify quorum service for a particular quorum result that originated externally.
+ */
+public class QuorumCallbackImpl implements QuorumCallback {
+
+    private AtomicBoolean isFirstQuorumDone = new AtomicBoolean(false);
+    private AtomicBoolean quorumResult = new AtomicBoolean(true);
+    private NodeEngineImpl nodeEngine;
+    private String quorumName;
+
+    public QuorumCallbackImpl(NodeEngineImpl nodeEngine, String quorumName) {
+        this.nodeEngine = nodeEngine;
+        this.quorumName = quorumName;
+    }
+
+    public boolean isInitialized() {
+        return isFirstQuorumDone.get();
+    }
+
+    @Override
+    public boolean getResult() {
+        return quorumResult.get();
+    }
+
+    @Override
+    public void setResult(boolean quorumResult) {
+        setLocalResult(quorumResult);
+    }
+
+    @Override
+    public void setResultOnAllMembers(boolean quorumResult) {
+        setLocalResult(quorumResult);
+        sendResultToMembers(quorumResult);
+    }
+
+    private void setLocalResult(boolean quorumResult) {
+        this.isFirstQuorumDone.set(true);
+        this.quorumResult.set(quorumResult);
+    }
+
+    private void sendResultToMembers(boolean quorumResult) {
+        OperationService operationService = nodeEngine.getOperationService();
```

Comment:
Dumb question; why do the members need to know that the quorum state changed? If each member listens to the member set, then it will always know if there are sufficient members for a quorum or not. Imho this only increases risk e.g. if you can't update a member e.g. because of a splitbrain or due to a slow system.